controllers/sriovnetworknodepolicy_controller.go

```diff
@@ -308,6 +307,41 @@ func (r *SriovNetworkNodePolicyReconciler) syncAllSriovNetworkNodeStates(ctx con
 	return nil
 }
 
+// handleStaleNodeState handles stale SriovNetworkNodeState CR (the CR which no longer have a corresponding node with the daemon).
+// If the CR has the "keep until time" annotation, indicating the earliest time the state object can be removed,
+// this function will compare it to the current time to determine if deletion is permissible and do deletion if allowed.
+// If the annotation is absent, the function will create one with a timestamp in future, using either the default or a configured offset.
+func (r *SriovNetworkNodePolicyReconciler) handleStaleNodeState(ctx context.Context, ns *sriovnetworkv1.SriovNetworkNodeState) error {
+	logger := log.Log.WithName("handleStaleNodeState")
+	now := time.Now().UTC()
+	keepUntilTime := ns.GetKeepUntilTime()
+	if keepUntilTime.IsZero() {
+		// keep until time annotation is not set, set a new value with default or configured offset and update the object
+		delayMinutes, err := strconv.Atoi(os.Getenv("STALE_NODE_STATE_CLEANUP_DELAY"))
+		if err != nil || delayMinutes <= 0 {
+			delayMinutes = 30 // keep objects for 30 minutes by default
```

Comment:
Let's add some logging here to indicate env variable is incorrect

Reply:
nit: we've got unused CleanupTimeout [1] variable. maybe we need to change it's value and use it here

[1] https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/68b6c020fbba1596ddd3f2e4a70cfcdb034b7e28/test/util/util.go#L34

Reply:
I added constant